### PR TITLE
Allows Roller beds to be converted into surgery tables with a wrench.

### PR DESCRIPTION
**Pull request by hyper2snyper**
## About The Pull Request

Creates the capability in roller beds to be able to set them up as a surgery table, allowing all surgery table operations with it.
Using a wrench on a deployed roller bed with activate its surgery table mode. Once deployed it can be turned back into a portable roller bed with a wrench.
In surgery mode it can do all usual surgery table operations.

## Why It's Good For The Game

It will allow field surgery to be far less tedious as it allows surgery to be set up almost anywhere.
I can see how this could cause some balance problems since you wouldn't need shipside as much, however surgery is often not required for brining marines back to life in general.
Open to suggestions about how to balance this if it is needed.

> Powercreep. Allows surgery to be done anywhere with no fail chance from surface.

Surgery in general is easy, and it is mainly done shipside and on the Alamo if the Op table has been installed.
Surgery is a long term fix to something that is almost always fixed shorterm, the main exception I can think of is removing larva.
All of these are done in surgery a few second however take anywhere from 30 seconds to 5 minutes depending on how often the Alamo is cycled.
My aim here is to make fixing things like fractures not require the Marine to stop playing for 5 minutes as they wait for transit, and to allow a MO or CMO to not have to leave the planet to tend marines.
For CMOs and MOs it should make setting up FoB surgery far easier, and for Corpsmen it would require them to carry all the extra equipment that is required for setting up the bed and performing surgery, not to mention it is immobile in that state and would most likely cause the marine and corpsmen to be vunerable.

An alternative I have thought about was making a few special rollers given to MOs and CMOs for this purpose, however most of their surplus medical gear is given out roundstart anyways so I don't think it would cause too much of a difference. 

## Changelog
:cl:
add: Allows Roller beds to be converted into surgery tables with a wrench.
/:cl:

### CODE REVIEW

**Comment by bobmancer343** [comment]
bad

File: code/game/objects/machinery/OpTable.dm [hunk 1]
```diff
@@ -230,3 +227,37 @@
...
 
 	return 1
+
+/obj/machinery/optable/update_icon_state() //This is to simplify and clean some above code.
+	. = ..()
+	if(!victim)
+		icon_state = initial(icon_state)
+		return
+	if(victim.handle_pulse())
+		icon_state = initial(icon_state) + "-active"
```
Comment (BraveMole) [inline code/game/objects/machinery/OpTable.dm:237]: You missed an early return it seems. Also this could be made in one if return, not two

Comment (BraveMole) [reply code/game/objects/machinery/OpTable.dm:237]: if (victim?handle_pulse())
    icon = thing
    return
icon = other thing

File: code/game/objects/machinery/OpTable.dm [hunk 2]
```diff
@@ -230,3 +227,37 @@
...
+			to_chat(user, "<span class='notice'>You cannot configure the [src] to roller bed with [victim] still buckled.</span>")
+			return
+		if(anes_tank)
+			to_chat(user, "<span class='notice'>You cannot configure the [src] to roller bed with the [anes_tank] still attached.</span>")
+			return
+		playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
+		new rollertype(loc)
+		qdel(src)
+		return
+	. = ..()
```
Comment (BraveMole) [inline code/game/objects/machinery/OpTable.dm:261]:
```suggestion
	..()
```

File: code/game/objects/machinery/OpTable.dm [hunk 3]
```diff
@@ -230,3 +227,37 @@
...
+	icon_state = initial(icon_state)
+
+
+/obj/machinery/optable/rollerop
+	name = "portable surgery table"
+	desc = "A roller bed in surgery table mode. It is unmovable yet capable of all regular surgery table operations. Use a wrench to return it to a portable state."
+	icon = 'icons/obj/rollerbed.dmi' 
+	icon_state = "roller_surge"
+	use_power = NO_POWER_USE
+	var/rollertype = /obj/structure/bed/roller
```
Comment (BraveMole) [inline code/game/objects/machinery/OpTable.dm:247]: autodoc

File: code/game/objects/machinery/OpTable.dm [hunk 4]
```diff
@@ -230,3 +227,37 @@
...
+	. = ..()
+	if(!victim)
+		icon_state = initial(icon_state)
+		return
+	if(victim.handle_pulse())
+		icon_state = initial(icon_state) + "-active"
+	icon_state = initial(icon_state)
+
+
+/obj/machinery/optable/rollerop
```
Comment (BraveMole) [inline code/game/objects/machinery/OpTable.dm:241]: This has no drawbacks compared to a regular optable? Should it not have some?

File: code/game/objects/machinery/OpTable.dm (hunk 1 above)
Comment (BraveMole) [reply code/game/objects/machinery/OpTable.dm:237]: And anyway, this doesn't work.

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by Chaznoodles** [comment]
Powercreep. Allows surgery to be done anywhere with no fail chance from surface.

**Comment by TGCMbetcher** [comment]
> Powercreep. Allows surgery to be done anywhere with no fail chance from surface.

Ye I was thinking a req item to do that, the only difference is the anesthetic part so something to do with that.

**Comment by hyper2snyper** [comment]
> Powercreep. Allows surgery to be done anywhere with no fail chance from surface.

Yes I can see that quite obviously however.

Surgery in general is easy, and it is mainly done shipside and on the Alamo if the Op table has been installed.
Surgery is a long term fix to something that is almost always fixed shorterm, the main exception I can think of is removing larva.
All of these are done in surgery a few second however take anywhere from 30 seconds to 5 minutes depending on how often the Alamo is cycled.
My aim here is to make fixing things like fractures not require the Marine to stop playing for 5 minutes as they wait for transit, and to allow a MO or CMO to not have to leave the planet to tend marines.
For CMOs and MOs it should make setting up FoB surgery far easier, and for Corpsmen it would require them to carry all the extra equipment that is required for setting up the bed and performing surgery, not to mention it is immobile in that state and would most likely cause the marine and corpsmen to be vunerable.

An alternative I have thought about was making a few special rollers given to MOs and CMOs for this purpose, however most of their surplus medical gear is given out roundstart anyways so I don't think it would cause too much of a difference.

**Comment by Walarks** [comment]
You could balance it around requiring electricity and batteries to stay active.

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by hyper2snyper** [comment]
Closing because of good discussion in #dev_general

**Pull request closed without merging**